package/scripts/presto_coordinator.py

```diff
@@ -39,17 +42,22 @@ def start(self, env):
         Execute('{0} start'.format(daemon_control_script))
         if 'presto_worker_hosts' in host_info.keys():
             all_hosts = host_info['presto_worker_hosts'] + \
-                host_info['presto_coordinator_hosts']
+                        host_info['presto_coordinator_hosts']
         else:
             all_hosts = host_info['presto_coordinator_hosts']
-        smoketest_presto(
-            PrestoClient('localhost','root',
-                         config_properties['http-server.http.port']),
-            all_hosts)
+        # sleep some seconds for waiting coordinator start finished
+        time.sleep(30)
+        smoketest_presto(PrestoClient('localhost', 'root', config_properties['http-server.http.port']), all_hosts)
 
     def status(self, env):
         from params import daemon_control_script
-        Execute('{0} status'.format(daemon_control_script))
+        try:
+            Execute('{0} status'.format(daemon_control_script))
+        except ExecutionFailed as ef:
+            if ef.code == 3:
+                raise ComponentIsNotRunning("ComponentIsNotRunning")
```

Comment:
Yes. Ambari only accept zero value as normal return, any other return value will cause an ExecutionFailed exception thrown. As `presto status` will return the code: 3 .

Also In Ambari 2.5, Ambari will handle ComponentIsNotRunning exception as a normal logic process like below:
/usr/lib/python2.6/site-packages/resource_management/libraries/script/script.py (Ambari Libaray):

> service_name = config['serviceName'] if config is not None and 'serviceName' in config else None
      try:
        #TODO Once the logic for pid is available from Ranger and Ranger KMS code, will remove the below if block.
        services_to_skip = ['RANGER', 'RANGER_KMS']
        if service_name in services_to_skip:
          Logger.info('Temporarily skipping status check for {0} service only.'.format(service_name))
        elif is_stack_upgrade:
          Logger.info('Skipping status check for {0} service during upgrade'.format(service_name))
        else:
          self.status(env)
          raise Fail("Stop command finished but process keep running.")
      except ComponentIsNotRunning as e:
        pass  # expected
      except ClientComponentHasNoStatus as e:
        pass  # expected

      # To remain backward compatible with older stacks, only pass upgrade_type if available.
      # TODO, remove checking the argspec for "upgrade_type" once all of the services support that optional param.
      self.pre_start()
      if "upgrade_type" in inspect.getargspec(self.start).args:
        self.start(env, upgrade_type=upgrade_type)
      else:
        if is_stack_upgrade:
          self.start(env, rolling_restart=(restart_type == "rolling_upgrade"))
        else:
          self.start(env)

So. TO AVOID THE EXCEPTION 

> raise Fail("Stop command finished but process keep running.")

raise ComponentIsNotRunning exception should be the right way.